gtsam/discrete/DiscreteFactor.h

```diff
@@ -129,6 +129,16 @@ class GTSAM_EXPORT DiscreteFactor : public Factor {
   /// DecisionTreeFactor
   virtual DecisionTreeFactor operator*(const DecisionTreeFactor&) const = 0;
 
+  /**
+   * @brief Multiply in a DiscreteFactor and return the result as
+   * DiscreteFactor, both via shared pointers.
+   *
+   * @param df DiscreteFactor shared_ptr
+   * @return DiscreteFactor::shared_ptr
+   */
+  virtual DiscreteFactor::shared_ptr multiply(
```

Comment:
I feel like this can be `DiscreteFactor::shared_ptr operator*(const DiscreteFactor::shared_ptr&)` instead of `multiply`.

Reply:
No, operator* does not normally do shared pointers

Reply:
Gotcha